src/docgen/schema.ts

```diff
@@ -0,0 +1,335 @@
+/**
+ * Describes a type.
+ */
+export interface TypeSchema {
+
+  /**
+   * The type FQN. If missing, the type is a reference (array, map..)
```

Comment:
...or a primitive?

Is there a reason this type isn't a discriminated union? Feels like it would be much more clear to consume if there was a type field going `type: primitive`, `type: referenceType`, `type: collectionType` etc.

Reply:
Thinking about it some more... at this point it looks like the schema is really more of a 1:1 mirror of the jsii assembly. We're losing *some* information but not a lot.

If we're going to retain most of the information from the jsii assembly, then feels like we might as well render from that?

I think the challenge is to lose and compact as much information as we can, while still being able to produce the output that we need. At that point we'll have found the essence of this data model.

I reckon it would maybe looks something a little more like this?

```ts
interface Type {
  id: string;
  name: string;
  jsiiFqn: string;
  summary: string;
  details: string;
  annotations: Annotation[]; // This is where we would stick "deprecated", "experimental", etc
  memberGroups: MemberGroup[]; // Properties, methods, constructors, ...
  // etc
}

interface MemberGroup {
  caption: string;
  members: Member[];
}

interface TypeReference {
  types: SingleType[];
}

interface SingleTypeReference {
  displayName: string;
  link?: Link;
}

type Link =
  | { type: 'reference', id: string }
  | { type: 'href', href: string };


```

Reply:
And if we say "no we want it to be mostly the jsii assembly schema but with less data", then feels like the definitions could be a set of `Omit<>`/`Pick<>`-based mapped types...?

Reply:
so I do agree with this in principle, and right now it is very close to the assembly, but the main point is that this content is already language specific. So the consumer is avoiding doing all the language specific translations coming from the assembly directly. I see the purpose of the "view model" aka, the next level of translation that we have discussed to be the place where we want to "lose" as much information and present it in a format that can dictate the layout of the content.